include/boost/thread/pthread/mutex.hpp

```diff
@@ -318,7 +318,7 @@ namespace boost
           Duration d = t - Clock::now();
           if ( d <= Duration::zero() ) return false;
           d = (std::min)(d, Duration(milliseconds(100)));
-          while ( ! try_lock_until(thread_detail::internal_clock_t::now() + ceil<nanoseconds>(d)))
+          while ( ! try_lock_until(thread_detail::internal_clock_t::now() + d))
```

Comment:
Why not ceil here?

Reply:
The result of `thread_detail::internal_clock_t::now() + d` will be whatever type is common to both `internal_clock_t::duration` and `Duration`. Since `try_lock_until()` will accept any duration type, converting to nanoseconds is unnecessary. Does that make sense?

Reply:
yes :)